packages/admin-next/dashboard/src/v2-routes/taxes/tax-province-create/components/tax-region-create-form.tsx

```diff
@@ -0,0 +1,241 @@
+import { zodResolver } from "@hookform/resolvers/zod"
+import { Button, Heading, Input, Select, Switch, Text } from "@medusajs/ui"
+import { useForm } from "react-hook-form"
+import { useTranslation } from "react-i18next"
+import * as zod from "zod"
+
+import { TaxRegionResponse } from "@medusajs/types"
+import { Form } from "../../../../components/common/form"
+import { PercentageInput } from "../../../../components/common/percentage-input"
+import {
+  RouteFocusModal,
+  useRouteModal,
+} from "../../../../components/route-modal"
+import { useCreateTaxRegion } from "../../../../hooks/api/tax-regions"
+import { countries } from "../../../../lib/countries"
+
+export const TaxRegionCreateForm = ({
+  taxRegion,
+  formSchema,
+}: {
+  taxRegion?: TaxRegionResponse
+  formSchema: zod.ZodObject<{
+    province_code: any
+    country_code: any
+    parent_id: any
+    name: any
+    code: any
+    rate: any
+    is_combinable: any
+  }>
+}) => {
+  const { t } = useTranslation()
+  const { handleSuccess } = useRouteModal()
+
+  const form = useForm<zod.infer<typeof formSchema>>({
+    defaultValues: {
+      country_code: taxRegion?.country_code || undefined,
+      parent_id: taxRegion?.id || undefined,
+    },
+    resolver: zodResolver(formSchema),
+  })
+
+  const { mutateAsync, isPending } = useCreateTaxRegion()
+
+  const handleSubmit = form.handleSubmit(async (data) => {
+    await mutateAsync(
+      {
+        parent_id: taxRegion?.id,
+        province_code: data.province_code,
+        country_code: data.country_code,
+        default_tax_rate: {
+          name: data.name,
+          code: data.code,
+          rate: data.rate,
+        },
+      },
+      {
+        onSuccess: () =>
+          taxRegion?.id
+            ? handleSuccess(`/settings/taxes/${taxRegion.id}`)
+            : handleSuccess(`/settings/taxes`),
+      }
+    )
+  })
+
+  return (
+    <RouteFocusModal.Form
+      form={form}
+      className="flex h-full flex-col overflow-hidden"
+    >
+      <form onSubmit={handleSubmit} className="flex h-full flex-col">
+        <RouteFocusModal.Header>
+          <div className="flex items-center justify-end gap-x-2">
+            <RouteFocusModal.Close asChild>
+              <Button variant="secondary" size="small">
+                {t("actions.cancel")}
+              </Button>
+            </RouteFocusModal.Close>
+
+            <Button
+              type="submit"
+              variant="primary"
+              size="small"
+              isLoading={isPending}
+            >
+              {t("actions.create")}
+            </Button>
+          </div>
+        </RouteFocusModal.Header>
+
+        <RouteFocusModal.Body className="flex h-full w-full flex-col items-center overflow-hidden p-16">
+          <div className="flex flex-col gap-y-8 min-w-[400px]">
```

Comment:
Needs to be aligned with styling of FocusModal in the rest of admin

```suggestion
          <div className="flex flex-col gap-y-8 w-full max-w-[720px]">
```